src/components/explore-view.tsx

```
@@ -146,16 +141,18 @@ export function ExploreView({
-      // remaining category's rows too — those are always null, so the query
-      // would return nothing instead of just ignoring the unrelated filter.
+      // Gated on isClimbingActive: a grade filter with climbing off would force
+      // climbing_grade to match non-climbing rows too (always null) → zero results.
       climbingGrades:
         isClimbingActive && climbingGrades.size > 0 ? Array.from(climbingGrades) : undefined,
     }),
     [sizeClasses, amenities, wheelchairAccessibleOnly, climbingGrades, isClimbingActive]
   );
 
+  useEffect(() => {
+    if (!isHeatmapMode) setLegendDismissed(false);
```

ESLint

[error] Error: Calling setState synchronously within an effect can trigger cascading renders

Effects are intended to synchronize state between React and external systems such as manually updating the DOM, state management libraries, or other platform APIs. In general, the body of an effect should do one or both of the following:
* Update external systems with the latest state from React.
* Subscribe for updates from some external system, calling setState in a callback function when external state changes.

Calling setState synchronously within an effect body causes cascading renders that can hurt performance, and is not recommended. (https://react.dev/learn/you-might-not-need-an-effect).

/home/jailuser/git/src/components/explore-view.tsx:153:25
  151 |
  152 |   useEffect(() => {
> 153 |     if (!isHeatmapMode) setLegendDismissed(false);
      |                         ^^^^^^^^^^^^^^^^^^ Avoid calling setState() directly within an effect
  154 |   }, [isHeatmapMode]);
  155 |
  156 |   const selectedVibe = VIBE_OPTIONS.find((option) => option.value === activeVibe);

(react-hooks/set-state-in-effect)